Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/DriverFeeUpdates/DriverFee.hs

```diff
@@ -0,0 +1,121 @@
+        case mbDriver of
+          Nothing -> do
+            logInfo "Driver Not found. This should not be possible."
+            throwError (InternalError "Driver Not Found") -- Unreachable
+          Just driver -> do
+            let paymentTitle = "Payment Pending"
+                paymentMessage = "Payment of Rs. " <> show driverFee.totalAmount.getMoney <> " is pending. Pay by " <> show driverFee.payBy <> " to avoid being blocked."
+            (Notify.sendNotificationToDriver driver.merchantId FCM.SHOW Nothing FCM.PAYMENT_PENDING paymentTitle paymentMessage driver.id driver.deviceToken) `C.catchAll` \e -> C.mask_ $ logError $ "FCM for payment reminder to driver id " <> driver.id.getId <> " failed. Error: " <> show e
+      Esq.runNoTransaction $ mapM_ (updateStatus PAYMENT_PENDING) driverFeeIds
+      case listToMaybe driverFees of
+        Nothing -> return Complete
+        Just driverFee -> do
+          driver <- Esq.runInReplica $ QPerson.findById (cast driverFee.driverId) >>= fromMaybeM (PersonDoesNotExist driverFee.driverId.getId)
+          transporterConfig <- SCT.findByMerchantId driver.merchantId >>= fromMaybeM (TransporterConfigNotFound driver.merchantId.getId)
+          ReSchedule <$> getRescheduledTime transporterConfig
+    else do
+      logInfo "No ongoing payment found."
+      return Complete
+
+unsubscribeDriverForPaymentOverdue ::
+  ( MonadFlow m,
+    HedisFlow m r,
+    CoreMetrics m,
+    HasCacheConfig r,
+    EsqDBFlow m r,
+    Esq.EsqDBReplicaFlow m r
+  ) =>
+  Job 'UnsubscribeDriverForPaymentOverdue ->
+  m ExecutionResult
+unsubscribeDriverForPaymentOverdue Job {id, jobInfo} = withLogTag ("JobId-" <> id.getId) do
+  let jobData = jobInfo.jobData
+      startTime = jobData.startTime
+  now <- getCurrentTime
+  (driverFees, driverFeeIds, feeZipDriver, relevantDriverIds) <- calcDriverFeeAttr PAYMENT_PENDING startTime now
+  if not $ null driverFees
+    then do
+      for_ feeZipDriver $ \(driverFee, mbDriver) -> do
+        case mbDriver of
+          Nothing -> do
+            logInfo "Driver Not found. This should not be possible."
+            throwError (InternalError "Driver Not Found") -- Unreachable
+          Just driver -> do
+            let paymentTitle = "Blocked - Payment Overdue"
+                paymentMessage = "Sorry, you are blocked from taking rides as payment of Rs. " <> show driverFee.totalAmount.getMoney <> " is outstanding. Please pay the balance to get unblocked."
+            (Notify.sendNotificationToDriver driver.merchantId FCM.SHOW Nothing FCM.PAYMENT_OVERDUE paymentTitle paymentMessage driver.id driver.deviceToken) `C.catchAll` \e -> C.mask_ $ logError $ "FCM for removing subsciption of driver id " <> driver.id.getId <> " failed. Error: " <> show e
+      Esq.runTransaction $ mapM_ (updateStatus PAYMENT_OVERDUE) driverFeeIds
+      Esq.runTransaction $ mapM_ (updateSubscription False) relevantDriverIds
+    else do
+      logInfo "No ongoing payment found."
+  return Complete
```

Comment:
we are not `ReScheduling` this ?

Reply:
No, unsubscription happens only once

Reply:
okay we can discuss this later, for now its fine.